hack/hooks/pre-commit

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+
+echo
+echo "=== checking license info ==="
+make license/check
+
+echo
+echo "=== linting ==="
+make lint
+
+echo
+echo "=== testing ==="
+make test
+
+echo "=== building ==="
+make build
```

Comment:
a pre-commit should not build..

Reply:
hmm the idea was to be sure that I'm not commiting something that doesn't build (it is possible to pass tests + linter but build is broken, but it is rare).